Tests/OptimizelyTests-Common/OptimizelyUserContextTests.swift

```diff
@@ -0,0 +1,157 @@
+/****************************************************************************
+* Copyright 2020, Optimizely, Inc. and contributors                        *
+*                                                                          *
+* Licensed under the Apache License, Version 2.0 (the "License");          *
+* you may not use this file except in compliance with the License.         *
+* You may obtain a copy of the License at                                  *
+*                                                                          *
+*    http://www.apache.org/licenses/LICENSE-2.0                            *
+*                                                                          *
+* Unless required by applicable law or agreed to in writing, software      *
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+    
+
+import XCTest
+
+class OptimizelyUserContextTests: XCTestCase {
+    
+    let exp1Key = "exp_with_audience"
+    let var1AKey = "a"
+    let var1BKey = "b"
+    let exp1Id = "10390977673"
+    let var1AId = "10389729780"
+    let var1BId = "10416523121"
+
+    let exp2Key = "exp_no_audience"
+    let var2AKey = "variation_with_traffic"
+    let var2BKey = "variation_no_traffic"
+    let exp2Id = "10420810910"
+    let var2AId = "10418551353"
+    let var2BId = "10418510624"
+
+    let expUserId = "1234"
+    let expUserId2 = "3456"
+    let expOptimizely = OptimizelyClient(sdkKey: "any")
+
+    func testOptimizelyUserContext_userId() {
+        let user = OptimizelyUserContext(optimizely: expOptimizely, userId: expUserId)
+        
+        XCTAssert(user.optimizely == expOptimizely)
+        XCTAssert(user.userId == expUserId)
+        XCTAssert(user.attributes.count == 0)
+    }
+    
+    func testOptimizelyUserContext_nilAttributes() {
+        let user = OptimizelyUserContext(optimizely: expOptimizely, userId: expUserId, attributes: nil)
+        
+        XCTAssert(user.optimizely == expOptimizely)
+        XCTAssert(user.userId == expUserId)
+        XCTAssert(user.attributes.count == 0)
+    }
+
+    func testOptimizelyUserContext_attributes() {
```

Comment:
Please add one more test, when you modify existing key/value already in the attributes provided at the creation time.